sequencers/modeshape-sequencer-epub/src/main/java/org/modeshape/sequencer/epub/EpubMetadata.java

```diff
@@ -0,0 +1,346 @@
+
+import java.io.BufferedOutputStream;
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipInputStream;
+
+import javax.xml.namespace.NamespaceContext;
+import javax.xml.parsers.DocumentBuilder;
+import javax.xml.parsers.DocumentBuilderFactory;
+import javax.xml.xpath.XPath;
+import javax.xml.xpath.XPathConstants;
+import javax.xml.xpath.XPathExpression;
+import javax.xml.xpath.XPathFactory;
+
+import org.modeshape.common.util.IoUtil;
+import org.modeshape.sequencer.epub.EpubMetadataProperty.AlternateScript;
+import org.w3c.dom.Document;
+import org.w3c.dom.Node;
+import org.w3c.dom.NodeList;
+
+/**
+ * Utility for extracting Metadata from EPUB format.
+ */
+public class EpubMetadata {
+
+    static final String[] MIME_TYPE_STRINGS = { "application/epub+zip" };
+
+    // The XML namespace for the Dublin Core schema.
+    static final String DUBLIN_CORE_PREFIX = "dc";
+    static final String DUBLIN_CORE_URI = "http://purl.org/dc/elements/1.1/";
+
+    private List<EpubMetadataProperty> title = new ArrayList<>();
+    private List<EpubMetadataProperty> creator = new ArrayList<>();
+    private List<EpubMetadataProperty> contributor = new ArrayList<>();
+    private List<EpubMetadataProperty> language = new ArrayList<>();
+    private List<EpubMetadataProperty> identifier = new ArrayList<>();
+    private List<EpubMetadataProperty> description = new ArrayList<>();
+    private List<EpubMetadataProperty> publisher = new ArrayList<>();
+    private List<EpubMetadataProperty> rights = new ArrayList<>();
+    private List<EpubMetadataProperty> date = new ArrayList<>();
+    private List<EpubMetadataProperty> alternateScript = new ArrayList<>();
+
+    private InputStream in;
+
+    public EpubMetadata( InputStream inputStream ) {
+        this.in = inputStream;
+    }
+
+    /*
+     * Check that given file is supported by this sequencer.
+     * The sequencer first examines the container to get the list of all rootfiles.
+     * Then it process each rootfile and extracts the metadata.
+     */
+    public boolean check() throws Exception {
+        // we need to create a copy of the file, because the container file
+        // might be at the end of the stream.
+        File fileCopy = File.createTempFile("modeshape-sequencer-epub", ".tmp");
+        IoUtil.write(in, new BufferedOutputStream(new FileOutputStream(fileCopy)));
+
+        ZipInputStream zipStream = new ZipInputStream(new FileInputStream(fileCopy));
+        List<String> rootfiles = getRootfiles(zipStream);
+        zipStream.close();
+
+        zipStream = new ZipInputStream(new FileInputStream(fileCopy));
```

Comment:
same comment as above; in addition, perhaps the construction of the `zipStream` can be deferred to only when `!rootFiles.isEmpty()`. It seems it's not required in any other case.